.circleci/config.yml

```diff
@@ -153,34 +162,48 @@ commands:
           name: Update deployments sheet
           command: |
             set -u
-            TZ=America/New_York
-            DATE=`TZ=$TZ date +%F`
-            TIME=`TZ=$TZ date +%T`
-            COMPONENT='<<parameters.study_key>> Angular client'
-            BRANCH=<< pipeline.git.branch >>
-            # Picking up any tags attached to the current commit
-            TAG=`git tag --points-at HEAD | xargs`
-            # Extract email from current commit
-            AUTHOR=`git show -s --format='%ae' HEAD`
+            set +x
             readvault.sh secret/pepper/${ENVIRONMENT}/v1/conf .data.gcp.serviceKey | \
-            sheetappend $RELEASE_SHEET_ID $DATE $TIME "$COMPONENT" $ENVIRONMENT $BRANCH "$TAG" $SHORT_GIT_SHA "$AUTHOR" $CIRCLE_BUILD_NUM "$CIRCLE_BUILD_URL"
-
+            reportdeploy.sh "$RELEASE_SHEET_ID" '<<parameters.study_key>> Angular client' "$ENVIRONMENT" "$CIRCLE_BUILD_NUM" "$CIRCLE_BUILD_URL"
+      - run:
+          name: Delete previous versions of service
+          command: |
+            set -u
+            set +x
+            # extract the service name from the GAE yaml file
+            CONFIG_FILE="${ANGULAR_PROJECT_DIR_PATH}/app.yaml"
+            SERVICE=$(grep -Eo -m1 "^service:\s*.*" "${CONFIG_FILE}" | awk '{print $2}')
+            echo "The service name is *${SERVICE}*"
+
+            # find what other versions are currently running
+            # skip first line, exclude our version, print 2nd column
+            AWK_COMMMAND="NR > 1 && !/${SHORT_GIT_SHA}/ {print \$2}"
+            VERSIONS_TO_DELETE=$(gcloud app versions list --service "$SERVICE" --project "broad-ddp-${ENVIRONMENT}" | awk "${AWK_COMMMAND}")
+
+            for VERSION_TO_DELETE in $VERSIONS_TO_DELETE; do
+              echo "Deleting version $VERSION_TO_DELETE of service $SERVICE"
+              gcloud app versions delete --quiet --service "$SERVICE" "$VERSION_TO_DELETE" --project "broad-ddp-${ENVIRONMENT}"
+            done;
+          working_directory: ~/repo/ddp-workspace
```

Comment:
Doing same in java back-end side: delete service when new deployment occurs. There is a limit on the number of versions you can have in a GCP project .